File: src/main/java/br/com/rest/gerenciador_academico/model/Turma.java
package br.com.rest.gerenciador_academico.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import java.util.Set;

@Data
@Entity
public class Turma {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String nome;

    private int ano;
    private String periodo;

    @ManyToOne
    @JoinColumn(name = "professor_id", nullable = false)
    private Professor professor;

    @OneToMany(mappedBy = "turma", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private Set<Inscricao> inscricoes;
}

File: frontend/gerenciador-academico-frontend/src/interfaces/turma.ts
export interface Turma{
    id: number;
    ano: String;
    periodo: String;
    professor?:{
        id: number;
        nome: String;
    } | null;
    disciplina?:{
        id: number;
        nome: String;
    } | null;
    inscricoes?: any[];
}